Update PR Comments & Gates with limitations (#32304)
DataDog/documentation: content/en/integrations/guide/source-code-integration.md

### PR comments

<div class="alert alert-warning">

>   PR comments are not supported in pull requests in public repositories or that are targeting a destination branch that is in a different repository from the source branch (i.e. forked repositories trying to merge into the main repository).

Reviewer: ```suggestion
  PR comments are not supported in pull requests in public repositories, or on pull requests targeting a destination branch in a different repository from the source branch (such as forked repositories trying to merge into the main repository).
```
